topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -1004,10 +995,10 @@ private void updateStatusForException(ReconcilableTopic reconcilableTopic, Excep
         String message = e.getMessage();
         String reason;
         if (e instanceof TopicOperatorException) {
-            LOGGER.debugCr(reconcilableTopic.reconciliation(), "Updating status for exception {}", e.toString());
+            LOGGER.errorCr(reconcilableTopic.reconciliation(), "Reconciliation failed: {}", e.getMessage());
```

Comment:
`TopicOperatorException` covers a few cases, not all of which I would consider worthy of an error.

For example, a `ResourceConflict` is a perfectly normal thing for the operator to encounter. It does not indicate an error in the operator (which is what an `error` log level means to me), it indicates an error on behalf of the user, which they're informed about using the status. 

Likewise, it's perfectly normal, from the operator's PoV, to have Kafka-specific exceptions (so not like NPE) coming from the Admin client. Those do not indicate that there's any problem with the operator, but with Kafka or the connection to it.

So for me I think using error here to too much. If you think we need to have error logging for these things because some users use the logging rather than looking at the status then we should think about what log category we should use for that. I.e. it would be more clear to have an `error` in the `user-error` category than in the `BatchingTopicController` category, imho. But if you all think this is fine, then that's fine with me.